src/clients/SpokePoolClient.ts

```diff
@@ -706,19 +770,34 @@ export class SpokePoolClient extends BaseAbstractClient {
       }
     }
 
-    if (eventsToQuery.includes("FilledRelay")) {
-      const fillEvents = queryResults[eventsToQuery.indexOf("FilledRelay")];
+    if (eventsToQuery.includes("RequestedV3SlowFill")) {
+      const slowFillRequests = queryResults[eventsToQuery.indexOf("RequestedV3SlowFill")];
+      for (const event of slowFillRequests) {
+        const slowFillRequest: SlowFillRequestWithBlock = {
+          ...spreadEvent(event.args),
+          destinationChainId: this.chainId,
+        };
+        const relayDataHash = getRelayDataHash(slowFillRequest, this.chainId);
+        this.slowFillRequests[relayDataHash] = slowFillRequest;
+      }
+    }
+
+    if (eventsToQuery.includes("FilledRelay") || eventsToQuery.includes("FilledV3Relay")) {
+      const fillEvents = [
+        ...((queryResults[eventsToQuery.indexOf("FilledRelay")] ?? []) as FilledRelayEvent[]),
+        ...((queryResults[eventsToQuery.indexOf("FilledV3Relay")] ?? []) as FilledV3RelayEvent[]),
+      ];
 
       if (fillEvents.length > 0) {
         this.log("debug", `Using ${fillEvents.length} newly queried fill events for chain ${this.chainId}`, {
           earliestEvent: fillEvents[0].blockNumber,
         });
       }
+
       for (const event of fillEvents) {
-        const rawFill = spreadEventWithBlockNumber(event) as FillWithBlock;
-        const fill: FillWithBlock = {
-          ...rawFill,
-        };
+        const fill = this.isV3FillEvent(event)
+          ? { ...(spreadEventWithBlockNumber(event) as v3FillWithBlock) }
+          : { ...(spreadEventWithBlockNumber(event) as v2FillWithBlock) };
 
         assign(this.fills, [fill.originChainId], [fill]);
         assign(this.depositHashesToFills, [this.getDepositHash(fill)], [fill]);
```

Comment:
For v3, should we be storing fills according to the relay hash? At the contract level, v3 deposits, fills, slow fills all use the same V3RelayData to identifiy bridge txns